### PR TITLE
Add com.hostbuf.FinalShell

### PR DESCRIPTION
**Pull request by jcleng**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by jcleng** [comment]
bot, build com.hostbuf.FinalShell

**Comment by jcleng** [comment]
bot, build com.hostbuf.FinalShell

**Comment by jcleng** [comment]
bot, build com.hostbuf.FinalShell

File: com.hostbuf.FinalShell.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        {
+            "name": "install",
+            "builddir": true,
+            "buildsystem": "simple",
+            "build-commands": [
+                "install -Dm644 com.hostbuf.FinalShell.metainfo.xml -t $FLATPAK_DEST/share/metainfo",
+                "install -Dm644 com.hostbuf.FinalShell.desktop /app/share/applications/com.hostbuf.FinalShell.desktop",
+                "install -Dm644 com.hostbuf.FinalShell.png /app/share/icons/hicolor/scalable/apps/com.hostbuf.FinalShell.png",
+                "install -Dm644 com.hostbuf.FinalShell.png /app/share/icons/hicolor/512x512/apps/com.hostbuf.FinalShell.png",
+                "ls FinalShell",
```
Comment (hfiguiere) [inline com.hostbuf.FinalShell.json:43]:
```suggestion
```

File: com.hostbuf.FinalShell.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "23.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "FinalShell",
+    "finish-args": [
+        "--share=network",
+        "--share=ipc",
+        "--device=dri",
+        "--socket=wayland",
+        "--socket=x11",
```
Comment (hfiguiere) [inline com.hostbuf.FinalShell.json:12]: why do you need both? EIther it supports wayland and it's fallback-x11, or it doesn't and then there is no socket for it.

File: com.hostbuf.FinalShell.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<application>
+  <id type="desktop">com.hostbuf.FinalShell</id>
+  <name>FinalShell</name>
+  <summary>FinalShell ssh tools</summary>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>LicenseRef-proprietary=https://www.hostbuf.com/t/988.html</project_license>
```
Comment (hfiguiere) [inline com.hostbuf.FinalShell.metainfo.xml:7]: is redistribution allowed?

File: com.hostbuf.FinalShell.json [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        "--filesystem=home"
```
[the remark this answers is not in the document]
Comment (jcleng) [reply com.hostbuf.FinalShell.json:12]: yes, use home dir to upload user file

File: com.hostbuf.FinalShell.metainfo.xml (hunk 3 above)
Comment (jcleng) [reply com.hostbuf.FinalShell.metainfo.xml:7]: yes

**Comment by jcleng** [comment]
bot, build com.hostbuf.FinalShell

**Comment by jcleng** [comment]
> image
> 
> Doesn't look exactly right on my end.我这边看起来不太对劲。



> image
> 
> Doesn't look exactly right on my end.

looks like Missing Chinese fonts, I see it, I'll rework it later, wating...

**Comment by jcleng** [comment]
bot, build com.hostbuf.FinalShell

**Comment by jcleng** [comment]
@barthalion Re-test it, I added the cjk font, I can't test it on my side, because the system has its own Chinese font.

**Pull request closed without merging**